Info.go

```diff
@@ -303,6 +316,20 @@ func (di *DownloadInfo) SetDownloadUrl(dataType, dlURL string) {
 	di.MDLInfo[dataType].DownloadURL = dlURL
 }
 
+func (di *DownloadInfo) GetMimeType(dataType string) string {
+	di.MDLInfo[dataType].Lock()
+	defer di.MDLInfo[dataType].Unlock()
```

Comment:
Read functions should use `RLock()` and `RUnlock()`. While unlikely, they're meant to allow for multiple concurrent reads without blocking.